addon/model-data.js

```diff
@@ -428,42 +446,73 @@ export default class M3ModelData {
     }
   }
 
-  _changedKeys(updates) {
-    let changedKeys = [];
-
-    if (updates) {
-      let original, i, value, key;
-      let keys = Object.keys(updates);
-      let length = keys.length;
-      let hasAttrs = this.hasChangedAttributes();
-      let attrs;
-      if (hasAttrs) {
-        attrs = this._attributes;
-      }
+  /*
+    Filters keys, which have local changes in _attributes, because even their value on
+    the server has changed, their local value is not and no property notification should
+    be sent for them.
+    @method _filterChangedKeys
+    @private
+   */
+  _filterChangedKeys(changedKeys) {
```

Comment:
can we remove it?

Reply:
This is supposed to be called by `pushData` and `didCommit` to filter out keys, which have local changes, because `_mergeUpdates()` does not consider them when merging. It can be reworked to be part of the `_mergeUpdates()`.

Reply:
it looks like it's not called yet tests are green?  missing a test?

Reply:
previously it was not called; do we have a test covering this?